src/pages/docs/experimental-languages/ts-language-guide/rpc.mdx

```diff
@@ -1,6 +1,266 @@
+    export component2-api;
+}
+```
+
+Updating the `main.ts` component implementations is also needed based on the above.
+
+To initialize dependencies, use `npm install` in both component's directory.
+
+### Generate RPC client stub definition and component
+
+All components that we want to call from another component will require client stub definitions and components.
+
+Let's create one for component2 using `golem-cli stubgen build`:
+
+```shell
+$ golem-cli stubgen build --source-wit-root component2/wit --dest-wasm component2-stub/component2_stub.wasm --dest-wit-root component2-stub/wit
+```
+
+The above command will place the generated **WIT** definitions and **WASM** component into the `component2-stub` folder:
+
+```shell
+$ tree component2-stub
+component2-stub
+├── component2_stub.wasm   // stub component
+└── wit
+    ├── _stub.wit          // stub definition
+    └── deps
+        ├── ...
+        .
+        .
+        .
+```
+
+### Add the stub dependency to the caller project
+
+Now we add the generated stub definition to the caller project with the `golem-cli stubgen add-stub-dependency`
+
+```shell
+golem-cli stubgen add-stub-dependency --stub-wit-root component2-stub/wit  --dest-wit-root component1/wit --overwrite
+```
+
+After this step the stub definitions will be present in the `component1/wit/deps` directory:
+
+```shell
+$  tree component1/wit
+component1/wit
+├── component1.wit
+└── deps
+    .
+    .
+    .
+    ├── rpcdemo_component2         // component2 definition
+    │   └── component2.wit
+    ├── rpcdemo_component2-stub    // component2 stub definition
+    │   └── _stub.wit
+    .
+    .
+    .
+```
+
+### Import the stub definition in the caller world
+
+To make the stub definition available for binding generation we have to import it in the callers world. To do so we have to
+edit the `component1/wit/component1.wit` file:
+
+```wit
+package rpcdemo:component1;
+
+interface api {
+    // ...
+}
+
+world component1 {
+    // ... other imports
+
+    // Add the following import for the client stub:
+    import rpcdemo:component2-stub/stub-component2;
+
+    export api;
+}
+```
+
+### Regenerate bindings
+
+To actually be able to use the imported stub we regenerate our bindings, which can done in the `component1` directory with:
+
+```shell
+$ npm run stub
+# the above will execute: jco stubgen wit -o src/generated
+```
+
+### Creating the worker client
+
+The worker to be invoked is identified by an URI which consists of the **component ID** and the **worker name**.
+
+If the worker pointed by the URI does not exists, it is going to be created automatically and it inherits the environment variables of the caller.
+
+A common pattern is to use environment variables to configure the deployed component's **component ID** and use that to construct the URI. See the [defining components](/docs/go-language-guide/defining-components) page for more information about passing configuration values to workers.
```

Comment:
`/docs/go-language-guide/defining-components` is this proper link for TS ?

Reply:
https://github.com/golemcloud/docs/pull/78/commits/b02500d6f82a48e816d16e163243b2b9a1cfd92f